python/plugins/processing/algs/qgis/Heatmap.py

```diff
@@ -179,10 +213,12 @@ def processAlgorithm(self, parameters, context, feedback):
         total = 100.0 / layer.featureCount() if layer.featureCount() else 0
         for current, f in enumerate(features):
             if kde.addFeature(f) != QgsKernelDensityEstimation.Success:
```

Comment:
Also - better add a check for feedback.isCanceled() here too and break out of the loop